pkg/metrics/metrics.go

```diff
@@ -106,7 +106,7 @@ var (
 	)
 )
 
-func MustRegister() {
+func init() {
```

Comment:
I wasn't clear on why we were not doing the `init` directly in this file. If there's a reason to avoid doing it here, I'm open to that as well.

Reply:
I think if we're going to wrap `MustRegister()` we would want to take input for the metrics we want to register (like we're passing onto the controller-runtime call), so a "always run this thing in a package" with no args is better suited to an init() call. I don't love falling back to init for these things but I see this pattern in other projects as well, cf:

- https://github.com/kubernetes-sigs/cluster-api/blob/v1.4.2/internal/runtime/metrics/metrics.go#L34-L38

That's not exactly what you asked, but given that this seems to be the one exception in this project (the other 7 packages that register metrics do so via `func init()`) this change makes sense.